darwin/future/meta/objects/dataset.py

```diff
@@ -155,3 +178,20 @@ def _validate_slug(slug: str) -> None:
 
         VALID_SLUG_CHARS = "abcdefghijklmnopqrstuvwxyz0123456789-_"
         assert_is(all(c in VALID_SLUG_CHARS for c in slug_copy), "slug must only contain valid characters")
+
+    def upload_files(
```

Comment:
Good move using the CLI function - I was looking at which to do when I went sick last week.